tests/Integration/Behaviour/Features/Context/Domain/Product/Combination/GenerateCombinationFeatureContext.php

```diff
@@ -51,6 +53,45 @@ public function generateCombinations(string $productReference, TableNode $table)
         ));
     }
 
+    /**
+     * @Then product :productReference default combination should be :combinationReference
+     *
+     * @param string $productReference
+     * @param string $combinationReference
+     */
+    public function assertCachedDefaultCombination(string $productReference, string $combinationReference): void
+    {
+        $this->assertCachedDefaultCombinationId(
+            $productReference,
+            $this->getSharedStorage()->get($combinationReference)
+        );
+    }
+
+    /**
+     * @Given product :productReference should not have a default combination
+     *
+     * @param string $productReference
+     */
+    public function assertProductHasNoCachedDefaultCombination(string $productReference): void
```

Comment:
> assertProductHasNoCachedDefaultCombination
Does that mean we could have a usecase where there is no "cached" default combination but database would have one?

I think (not sure) that Behat tests should ignore cache results 🤔  ?

Reply:
It is actually related to product->cache_default_attribute field in database.
This field is used as a shortcut to find current default combination id as well as determining if product is combinational

Reply:
Yes it's the database field name which is confusing